boa/src/context.rs

```diff
@@ -635,8 +636,15 @@ impl Context {
             .map_err(|e| e.to_string());
 
         let execution_result = match parsing_result {
-            Ok(statement_list) => statement_list.run(self),
-            Err(e) => self.throw_syntax_error(e),
+            Ok(statement_list) => {
+                // Once parsing has been successful build the realm
+                self.create_intrinsics();
+                statement_list.run(self)
+            }
+            Err(e) => Err(Value::String(RcString::from(format!(
+                "SyntaxError\": \"{}",
+                e
+            )))),
```

Comment:
we should not be throwing a string that contains `SyntaxError` we should throw an actual `SyntaxError` object

Reply:
I’ll have to look at how we can do that without requiring the realm. Right now the convenience method we have on the Context uses the realm so we can’t use that.

Reply:
@HalidOdat Yeah i think there's not much we can do here then, it wont be correct if we hard code `SyntaxError` and the code modifies it.